### ISSUE TITLE
Question regarding iPhone 7

### ISSUE THREAD

**Issue by andrewwiik** [kpwn/yalu102#400]
If i jailbreak my new iPhone 7 on 10.1.1 right now will I have issues using the new jailbreak for the iPhone 7 when it comes out? (the updated 10.1.1 one). Aka should i use the beta 3 one or wait for this updated one?

**Comment by XAVERRenogadeAngle** [comment]
#360 close this.

**Comment by NickGP** [comment]
@kpwn is going to eventually make an updated jailbreak for iPhone 7. However, I wouldn't recommend to download the beta since it isn't meant for iPhone 7 and the Yalu+Mach_portal isn't a complete jailbreak and can bring problems. I would just stay non jailbroken for now until something is updated for you.

**Comment by NickGP** [comment]
However, this isn't Code related at all. Next time try Reddit. #162 #64

**Comment by Platypus2** [comment]
Since this is probably going to be closed anyway, I'm curious as to what needs doing codewise to add support for iphone 7 in Yalu102. Is it a case of adding the memory protection exploits from Yalu1011 and merging them into yalu102 or is that way off? Wonder if it eventually be put on GitHub as I remember Luca saying he didn't want to opensource the memory protection stuff.

**Comment by NickGP** [comment]
It won't work because the KPP exploit used in the 10.2 jailbreak is patched on iPhone 7 but persists on all other 64bit devices

**Comment by Platypus2** [comment]
@NickGP Yes, sorry I should have been more specific. I know that it won't work on 10.2 - I meant 10.1.1, like the readme states it will support in the future. In theory if the exploits work 10.0-10.2, then the iPhone 7 with the additional exploits on 10.1.1 should work fine, I think that is the plan, but its not really been confirmed (that Yalu102 exploits will be used over trying to fix Ian Beer's exploit on 10.1.1), let alone the technical details before. If it has - it always gets lost in "iPhone 7 10.2 jb wen" comments :/

I'm just genuinely curious as to what is needed to support iPhone 7 on 10.1.1 given that the other 64 bit devices work fine on 10.1.1 afaik and whether or not it will be open source like the rest or bundled in separately later. Happy to just wait and see, but as this post popped up I thought I'd ask.

**Comment by andrewwiik** [comment]
My question wasn't when, it was regarding the compatability of the two....

**Comment by andrewwiik** [comment]
@NickGP this is code related as the incomapabilities would arise from different code in Yalu102 compared to Yalu b3

**Comment by XAVERRenogadeAngle** [comment]
I'm sure that kpwn is aware that plenty of people are already on beta 3 and will make things work properly.

**Comment by tangalbert919** [comment]
That might be true @XAVERRenogadeAngle, except you can't really be sure on that. People might've already upgraded to beta 7 because reboots are mean to semi-tethered jailbreaks (and I'm so glad @coolstar actually made a stashing tool that can replace the traditional stashing Cydia uses, making stashing theoretically safer).
@andrewwiik, Beta 3 should work on your iPhone 7, but the next update isn't until @kpwn gets his so-called "mixtape" working, which might take another day or two. Or an entire week. Don't go for the mach_portal+yalu jailbreak because it's unstable, and there's no chance of a gamma release for that jailbreak now.

**Comment by iphoneian** [comment]
It is for developers actually, if its your main phone then dont but if its about updating tweaks then go ahead.

**Comment by ghost** [comment]
Not code related issue, support has not been added yet. I'm sure Luca has an upgrade plan from mach_portal for iPhone 7.

#162

**Comment by blobbyblo** [comment]
This is code related but not in the way #162 and #64 intended the issues section to be.

But, there should be no problem switching between the two. Just like Pangu 9.3.3 users switching to Luca's safari jailbreak method. Different code, still works. I don't think you'll encounter issues switching between the two (as so far no 6s users have experienced issues).

**Comment by Platypus2** [comment]
6s users most likely clean restored to 10.2 though, I expect the majority of 6s users did not stay on 10.1.1, so any issues reported would be a minimum compared to that of iphone 7 users to come.

**Comment by andrewwiik** [comment]
@kpwn can you just settle this once and for all, can i use b3 on yalu+mach_portal and upgrade to yalu10 when supported without any issues or not?

**Comment by ghost** [comment]
#162

**Comment by ghost** [comment]
I'd wait. But #162 #64

**Comment by stevezaluk** [comment]
#162

**Comment by tangalbert919** [comment]
@zbduid12, that's not helping.

**Comment by stevezaluk** [comment]
@tangalbert919 well it isn't code related.

**Comment by tangalbert919** [comment]
Well, it doesn't hurt to explain why you referenced the issue in the first place. That might help prevent MORE issues like this in the future (if people bothered to read).
